openedx-unsupported/ecommerce
# ENT-638 Enterprise Offers

This PR adds the Enterprise offers feature. An Enterprise offer consists of a custom Condition model which checks that the user is linked to the specified EnterpriseCustomer, the course run(s) exist in that EnterpriseCustomer's catalogs, and the user has granted consent to share data for the course run(s).

Enterprise offers can be set up in the ecommerce administration tool.

ENT-638

**Testing Instructions:**
1. Visit [Enterprise Offers Admin](https://ecommerce-business.sandbox.edx.org/enterprise/offers) and verify that you can successfully create an Enterprise Offer using the existing [EnterpriseCustomerCatalogs](https://business.sandbox.edx.org/admin/enterprise/enterprisecustomercatalog/).
2. Visit https://business.sandbox.edx.org/enterprise/76afe510-a9fd-4b59-9ba1-49b537864205/course/course-v1:Microsoft+DAT206x+2017_T4/enroll/ as an unauthenticated user.
3. Log in through TestShib.
4. Verify that the course enrollment landing page displays the 50% Enterprise Offer discount.
5. Select Verified and click continue.
6. Verify the basket page displays the 50% Enterprise Offer discount.
